yarn-project/aztec-nr/aztec/src/oracle/get_secret_key.nr

```diff
@@ -1,17 +1,21 @@
-use crate::types::{
-    point::Point,
-    grumpkin_scalar::{
-        GRUMPKIN_SCALAR_SERIALISED_LEN,
-        GrumpkinScalar,
-        deserialise_grumpkin_scalar,
-    },
-};
 use crate::oracle::get_public_key::get_public_key;
+use crate::types::point::Point;
 
 #[oracle(getSecretKey)]
-fn get_secret_key_oracle(_owner: Point) -> [Field; GRUMPKIN_SCALAR_SERIALISED_LEN] {}
+fn get_secret_key_oracle(
+    _owner: Point,
+) -> [Field; dep::std::grumpkin_scalar::GRUMPKIN_SCALAR_SERIALISED_LEN] {
+}
 
-unconstrained fn get_secret_key(owner: Field) -> GrumpkinScalar {
-    let owner_nullifying_public_key = get_public_key(owner);
-    deserialise_grumpkin_scalar(get_secret_key_oracle(owner_nullifying_public_key))
-}
\ No newline at end of file
+unconstrained fn get_secret_key(owner: Field) -> dep::std::grumpkin_scalar::GrumpkinScalar {
```

Comment:
Heads up that all this logic is within an `unconstrained` function, meaning the assertions will not be enforced by the circuit (I'm wondering if this should be a compile error?). You probably want an unconstrained function that just wraps the oracle call, and then a non-unconstrained one that calls `get_public_key`, computes it from the secret, and runs the assertion (like in `get_public_key.nr`).